core/storage/trie/serialization/trie_serializer_impl.cpp

```diff
@@ -118,7 +88,13 @@ namespace kagome::storage::trie {
     if (db_key.empty() or db_key == getEmptyRootHash()) {
       return nullptr;
     }
-    OUTCOME_TRY(enc, backend_->load(db_key));
+    Buffer enc;
+    if (codec_->isMerkleHash(db_key)) {
```

Comment:
Here it's better to comment that a db key is actually the merkle value of the node, or better wrap Buffer in some MerkleValue type to ensure this contract on type level.